src/frontends/tensorflow/src/frontend.cpp

```diff
@@ -122,6 +126,18 @@ bool FrontEnd::supported_impl(const std::vector<ov::Any>& variants) const {
             // handle text protobuf format
             return true;
         }
+    } else if (variants[0].is<std::vector<std::string>>() && variants[0].as<std::vector<std::string>>().size() == 2) {
+        // here, we assume to get the input model path and checkpoints directory
+        auto paths = variants[0].as<std::vector<std::string>>();
+        auto model_path = paths[0];
+        auto checkpoints_dir = paths[1];
+        if (GraphIteratorProto::is_supported(model_path)) {
+            // binary protobuf format with checkpoints
+            return true;
+        } else if (GraphIteratorProtoTxt::is_supported(model_path)) {
+            // text protobuf format with checkpoints
+            return true;
+        }
```

Comment:
```suggestion
        return GraphIteratorProto::is_supported(model_path)) || GraphIteratorProtoTxt::is_supported(model_path));
```
nit: looks slightly cleaner, but it is just a visual preference.

Reply:
The current version looks to me more module-based, no mix:)